plutus-core/testlib/PlutusCore/Generators/QuickCheck/Builtin.hs

```diff
@@ -122,6 +126,26 @@ instance ArbitraryBuiltin ByteString where
     arbitraryBuiltin = Text.encodeUtf8 <$> arbitraryBuiltin
     shrinkBuiltin = map Text.encodeUtf8 . shrinkBuiltin . Text.decodeUtf8
 
+instance ArbitraryBuiltin BLS12_381.G1.Element where
+    arbitraryBuiltin =
+      BLS12_381.G1.hashToGroup <$> arbitrary <*> pure Data.ByteString.empty >>= \case
+      -- We should only get a failure if the second argument is greater than 255 bytes, which it isn't.
+           Left err -> error $ show err
+           Right p  -> pure p
+    shrinkBuiltin _ = []
```

Comment:
There's a zero element, right? Can we shrink to it, if it's not zero already? What about one? Worth some discussion at least.

Reply:
> There's a zero element, right? Can we shrink to it, if it's not zero already?

I guess so, although I'm not sure how meaningful that would be.  I don't think there's any sensible way to shrink these group elements in general: if we could come up with one it might even indicate some cryptographic weakness.

Reply:
OK, I've done that. I've forgotten how QuickCheck deals with shrinking though, so I'm still not sure if it makes.  What will happen if we get and error and the only thing we can shrink to is zero, which doesn't cause the error?  Does it just stop trying to shrink and give you back the thing that originally caused the problem?  I did get a non-terminating test because of something like this, but that may have been in Hedgehog.

Reply:
Oh.  This causes [prop_noTermShrinkLoops](https://github.com/input-output-hk/plutus/blob/924d1fcafa91da31a4a69097a48f159c6bc06b3a/plutus-core/plutus-ir/test/GeneratorSpec/Terms.hs#L175) to fail.

Reply:
I've just made all of the shrinks empty.